editor/src/components/canvas/controls/selection-area-helpers.ts

```diff
@@ -0,0 +1,132 @@
+    return 'scene'
+  }
+  return 'regular'
+}
+
+function elementIsFullyContainedInArea(
+  area: CanvasRectangle,
+  path: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+): boolean {
+  const element = MetadataUtils.findElementByElementPath(metadata, path)
+  return element == null || element.globalFrame == null || !isFiniteRectangle(element.globalFrame)
+    ? false
+    : rectangleContainsRectangle(area, element.globalFrame)
+}
+
+export const filterUnderSelectionArea = (
+  paths: ElementPath[],
+  metadata: ElementInstanceMetadataMap,
+  area: CanvasRectangle | null,
+  selectedViews: ElementPath[],
+): ElementPath[] => {
+  if (area == null) {
+    return []
+  }
+  const elements: ElementUnderSelectionArea[] = mapDropNulls((path) => {
+    const type = getElementUnderSelectionAreaType(metadata, path)
+    if (type == null) {
+      return null
+    }
+    return {
+      path: path,
+      type: type,
+      fullyContained: elementIsFullyContainedInArea(area, path, metadata),
+      selected: EP.containsPath(path, selectedViews),
+    }
+  }, paths)
+
+  const isSceneChild =
+    (element: ElementUnderSelectionArea) => (other: ElementUnderSelectionArea) => {
+      return other.type === 'scene' && EP.isDescendantOf(element.path, other.path)
+    }
+
+  const thereAreStoryboardChildren = elements.some(
+    (element) => element.type === 'regular' && !elements.some(isSceneChild(element)),
+  )
+
+  return elements
+    .filter((element) => {
+      // only outermost children
+      if (
+        element.type === 'regular' &&
+        elements.some((other) => {
+          return other.type !== 'scene' && EP.isDescendantOf(element.path, other.path)
+        })
+      ) {
+        return false
+      }
+
+      const parentScene = elements.find(isSceneChild(element))
+
+      // if the element is a schene child and there are storyboard children, skip it
+      if (parentScene != null && thereAreStoryboardChildren) {
+        return false
+      }
+      // if the element is a scene, and the scene is not fully contained skip the scene
+      if (element.type === 'scene' && !element.fullyContained && !element.selected) {
+        return false
+      }
+      // if a scene is fully contained, select just the scene and omit its children
+      if (parentScene != null && parentScene.fullyContained) {
+        return false
+      }
+      return true
+    })
+    .map((r) => r.path)
+}
+
+export function getSelectionAreaRenderedRect(
+  selectionArea: CanvasRectangle | null,
+  boundingRect: DOMRect | null,
+): CanvasRectangle | null {
+  if (selectionArea == null || boundingRect == null) {
+    return null
+  }
+  return canvasRectangle({
+    x: selectionArea.x - boundingRect.x,
+    y: selectionArea.y - boundingRect.y,
+    width: selectionArea.width,
+    height: selectionArea.height,
+  })
+}
+
+export function makeSelectionArea(from: WindowPoint, to: WindowPoint): CanvasRectangle {
```

Comment:
This function is converting from a `WindowPoint` to a `CanvasRectangle` without taking into account the canvas offset, meaning there are situations where the resultant rectangle will be at the incorrect position. From reading the code I'm not sure why this isn't causing issues with the area select when the canvas has been scrolled